gdk/wizard/Prompter.py

```diff
@@ -0,0 +1,259 @@
+            ):
+                return response
+            print(
+                f"Attempt {attempt}/{max_attempts}: Invalid response. Please try again.\nPlease vist: {link}"
+            )
+
+        if field == ConfigEnum.CUSTOM_BUILD_COMMAND:
+            self.utils.write_to_config_file(self.field_dict, self.project_config_file)
+            sys.exit(
+                "You have failed to enter a valid custom build command. Exiting wizard..."
+            )
+
+        print("Exceeded maximum attempts. Assuming current or default response.")
+        return current_field_value
+
+    def change_configuration(self, field_key, max_attempts=3):
+        """
+        Prompts the users to answer if they would like to change the build or publish configurations
+
+        Parameters
+        ----------
+            build_or_publish(string): takes value "build" or "publish"
+
+        Returns
+        -------
+            boolean: True if user wants to change the 'build_or_publish' configuration and False otherwise
+
+        """
+        self.parser.add_argument(
+            f"--change_{field_key}",
+            help=f"Change componenet {field_key} configurations",
+        )
+
+        for _ in range(max_attempts):
+            args = self.parser.parse_args(
+                [
+                    f"--change_{field_key}",
+                    input(
+                        f"Do you want to change the {field_key} configurations? (y/n) "
+                    ),
+                ]
+            )
+            response = getattr(args, f"change_{field_key}").strip().lower()
+            if response in {"y", "yes"}:
+                return True
+            elif response in {"n", "no"}:
+                return False
+            print("Your input was invalid response. Please respond again.")
+        return False
+
+    def interactive_prompt(self, field, value, require):
+        """
+        Prompts the user for a value of a given key through an interactive prompt.
+
+        Parameters
+        ----------
+            field (string): a field key of the gdk-config file to be changed
+            value (string): the current value corresponding the field key to be changed
+            require (boolean): if the field is a required field of the gdk config file
+
+        Returns
+        -------
+            string: the value of field that the user has entered through the interactive prompt
+        """
+        questions = [
+            {
+                "type": "input",
+                "name": "user_input",
+                "message": f"Current value of the {require}{field} is:",
+                "default": f"{value}",
+            }
+        ]
+        try:
+            answer = prompt(questions)
+            return answer["user_input"]
+        except (KeyError, TypeError):
+            self.utils.write_to_config_file(self.field_dict, self.project_config_file)
+            sys.exit("Wizard interrupted. Exiting...")
```

Comment:
raise Exception here.

Reply:
so instead of write_to_config_file and exiting wizard raise an Exception instead?